tensorboard/plugins/base_plugin.py

```diff
@@ -112,7 +113,10 @@ def __init__(
       db_module: A PEP-249 DB Module, e.g. sqlite3. This is useful for accessing
           things like date time constructors. This value will be None if we are
           not in SQL mode and multiplexer should be used instead.
-      logdir: The string logging directory TensorBoard was started with.
+      db_uri: The string db URI TensorBoard was started with. If this is set,
```

Comment:
IMO would be better to avoid too much argument proliferation.  Is there a way we could combine this with db_module and db_connection_provider somehow into like a DbInterface type that gets passed in?

Either that or we could replace logdir with something like data_uri or data_location here, and then interpret it as a logdir if the db_* arguments are not passed.

Reply:
I'm still thinking about this one. I see the problem, ie

1. `TBContext` is become sort of a [god object](https://en.wikipedia.org/wiki/God_object).
2. These 3 db-related properties are based on each other, so passing them separately is strange.

Would the DbInterface still be passed into `TBContext`? And `TBContext` thus gain say a `db_context` property containing those 3 properties? I'm trying to determine whether a new `DbInterface` parameter would simply the arguments needed here.

Reply:
SG. How about we create a subproperty for DB properties in a separate PR (as that would change several places in the code base).